### ISSUE TITLE
complete user sign up process

### ISSUE THREAD

**Issue by dottorer** [lale-help/lale-help#54]
We need to provide a complete onboarding processfor new users of the platform. This includes:

1. Differnetiation between sign up for new members and sign in for existing members

2. For sign up 
2.1 check for sufficient password strength (8chars, 1 num, one upper)
2.2 Ensure no duplicate emails or user nicknames
2.3 Capture all relevant user profile data in onboarding page flow
2.4 allow new user to select workgroups in the circle
2.5 notify circle and workgroup admins of new members with inApp message
2.6 enforce that user checks the 'I read the T&C nd data privacy rules' box and store results; allow clearing of that check by circle admin so that it needs to be re-checked during next login (in case T&C change or new members are invited by the admin)
2.7 validate email address with sending out email with 'complete verification' link

3. For sign in
3.1 provide 'forgot password' email to reset password
later possibly:
3.2 avoid ddos (I dislike captchas but something like that maybe)
3.3 enforce account freeze if wrong password entered 5 times?

In short, we need to provide an industry standard approach to handling sign up and sign in processes.

**Comment by dottorer** [comment]
@phil-monroe  added this issueas it seems important to handle user sign up process robustly as we engage with a wider audience. this should hopefully be pretty straight forward and not too hard to do as a process.

**Comment by phil-monroe** [comment]
Everything seems simple enough, but a couple quick notes:

2.7,3.1) Should be easy to add once we have mailing setup.

3.2) Cloudflare has a bunch of DOS/DDOS security checks in place, so I think we will be good for the short term. That being said, we can move to a CAPTCHA if we need to in the future.

**Comment by dottorer** [comment]
Missing from list is:
- validate password flow
- allow username as well as clear name
- important: track acceptance of T&Cs for site (link to doc on website) and storing consent in system.

Ideally, the signup should be represented as a mini wizard:
Step 1: enter your personal data (as is today)
Step 2: tell us about you (why are you joining as text field), tell us your skills from pick list.
Step 3: sign up for work groups (easy selection from existing work groups)

Then you are dropped onto the landing page which shows you open tasks right away for the picking.

**Comment by dottorer** [comment]
Missing items from the above list are:

1. verification of emails: #69 
2. offering pseudonym username: #121
3. new member approvals: #122 

Since we isolated the missing bits, I am closing this one and we are tracking bugs separately. Well done, @phil-monroe !
